docs: document Git Sync admin-only jobs and repository subresource access (#126807)
grafana/grafana: docs/sources/as-code/observability-as-code/git-sync/permissions-grafana.md

**Organization-level capabilities**:

- View dashboard preview links in pull requests
- Push their dashboard and folder changes to Git, including opening pull requests, via the jobs API

{{< admonition type="note" >}}

> Triggering a manual sync (**pull** from Git) and orphan-resource cleanup are admin-only operations. Editors can push their own changes but can't pull from Git on demand. Refer to [Job actions and required permissions](#job-actions-and-required-permissions).

Reviewer: ```suggestion
**Only Admins can trigger a manual sync (**pull** from Git) and orphan-resource clean-up**. Editors can push their own changes but can't pull from Git on demand. Refer to [Job actions and required permissions](#job-actions-and-required-permissions) for more details.
```

Users with the `Editor` role can manage sync operations but not infrastructure configuration:

| Permission Category  | Specific Permissions                                                                                              | What This Allows                                                                                             |
| -------------------- | ----------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------ |
| **Jobs**             | `provisioning.jobs:create`<br>`provisioning.jobs:read`<br>`provisioning.jobs:write`<br>`provisioning.jobs:delete` | Push changes and open pull requests<br>View sync jobs<br>Modify sync job settings<br>Cancel/delete sync jobs |
| **Read-Only Access** | `provisioning.repositories:read`<br>`provisioning.settings:read`                                                  | View repository configurations<br>View Git Sync settings                                                     |

{{< admonition type="note" >}}

> `provisioning.jobs:create` lets editors create jobs that push their changes (subject to the relevant `dashboards:*` and `folders:*` permissions), but it does **not** allow admin-only job actions such as a manual **pull** from Git or orphan-resource cleanup. Those require `provisioning.repositories:write`. Refer to [Job actions and required permissions](#job-actions-and-required-permissions).

Reviewer: ```suggestion
`provisioning.jobs:create` allows editors to create jobs to push their changes, subject to the relevant `dashboards:*` and `folders:*` permissions. However, **it doesn't allow admin-only job actions** such as a manual **pull** from Git or orphan-resource cleanup, which require the `provisioning.repositories:write` permission. Refer to [Job actions and required permissions](#job-actions-and-required-permissions) for more details.
```

- Dashboard-level permissions override folder-level permissions
- Changes made by users with appropriate permissions automatically sync to Git

### Job actions and required permissions

> Git Sync operations run as jobs. While `provisioning.jobs:create` is required to create any job, some job actions are additionally restricted to administrators and are gated on `provisioning.repositories:write` (an admin-only action). This prevents editors from triggering repository-wide operations even though they hold `provisioning.jobs:create`.

Reviewer: ```suggestion
Git Sync operations run as jobs. You need the `provisioning.jobs:create` permission to create any job. Moreover, **some job actions are restricted to administrators** and require the `provisioning.repositories:write` permission. This prevents editors from triggering repository-wide operations even though they hold `provisioning.jobs:create`.
```

| Job action                          | Required permission                                                                   | Who can run it                                            |
| ----------------------------------- | ------------------------------------------------------------------------------------- | --------------------------------------------------------- |
| Push changes / open pull request    | `provisioning.jobs:create` plus the relevant `dashboards:*` / `folders:*` permissions | Editors and Admins, on the resources they can modify      |
| Migrate resources                   | `provisioning.jobs:create` plus read/write on the affected resource types             | Editors and Admins with the required resource permissions |
| Manual sync (pull from Git)         | `provisioning.repositories:write`                                                     | Admins only                                               |
| Release / delete orphaned resources | `provisioning.repositories:write`                                                     | Admins only                                               |

### Repository subresource access

> The repository API exposes several subresources. The following table shows the permission each one is gated on. Because the `repositories` resource has no Editor tier (`repositories:read` is granted to Viewer and above, while `write`, `create`, and `delete` are admin-only), the `refs` subresource accepts either an admin (`repositories:write`) or an editor (`jobs:create`) check.

Reviewer 2: Do you think we should say **why** we accept either `repositories:write` or `jobs:create`? i.e. such subresource is used:
- When creating jobs by pushing changes to the repo
- When updating the `Repository` resource info

Author: let me make it more explicit.

Author: Good call — made it explicit in `8ae0c58`. The `refs` section now lists the two flows up front: editors pushing changes / opening PRs (authorized with `jobs:create`) and admins picking a target branch when setting up or editing a repository, which updates the `Repository` resource info (authorized with `repositories:write`). The "no Editor tier" rationale stays as the closing sentence.
